components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticator.java

```diff
@@ -0,0 +1,512 @@
+        }
+    }
+
+    /**
+     * This method is overridden to authenticate user.
+     *
+     * @param request  The request that is received by the authenticator.
+     * @param response The response that is received to the authenticator.
+     * @param context  The Authentication context received by authenticator.
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) {
+
+        String username = (String) context.getProperty(HYPR.USERNAME);
+
+        //Set the authenticated user.
+        AuthenticatedUser authenticatedUser =
+                AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(username);
+        context.setSubject(authenticatedUser);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Successfully logged in the user : " + username);
+        }
+    }
+
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context)
+            throws AuthenticationFailedException, LogoutFailedException {
+
+        if (context.isLogoutRequest()) {
+            // if the logout request comes, then no need to go through and complete the flow.
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+
+        } else if (request.getParameterMap().containsKey(HYPR.USERNAME)) {
+            // if the login form submission request comes, then go through this flow.
+            initiateHYPRAuthenticationRequest(request, response, context);
+            return AuthenticatorFlowStatus.INCOMPLETE;
+
+        } else if (context.getProperty(HYPR.AUTH_STATUS) != null) {
+            // if intermediate authentication request comes, then go through this flow.
+            String sessionDataKey = request.getParameter(HYPR.SESSION_DATA_KEY);
+            String authStatus = (String) context.getProperty(HYPR.AUTH_STATUS);
+
+            if (authStatus.equals(HYPR.AuthenticationStatus.COMPLETED.getName())) {
+                processAuthenticationResponse(request, response, context);
+                return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.PENDING.getName())) {
+                handleAuthenticationIncompleteState(response, sessionDataKey, HYPR.AuthenticationStatus.PENDING);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.CANCELED.getName())) {
+                handleAuthenticationIncompleteState(response, sessionDataKey, HYPR.AuthenticationStatus.CANCELED);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.FAILED.getName())) {
```

Comment:
let's switch this, and follow  <Constant>.equals(<variable>) format. Check in other places as well

Reply:
Fixed with [c32d436](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/c32d436550c773beca2b98f4586420ca1f8bc3c2)